File: Java/Laba8/src/main/java/ua/lviv/iot/goods/goods/Sketchbook.java
package ua.lviv.iot.goods.goods;

import ua.lviv.iot.goods.enums.CategoryEnum;

public class Sketchbook extends Good {

    private String colour;

    public Sketchbook(final Integer id, String name, CategoryEnum categoryEnum, boolean aviability, int price, String colour, int pages) {
        super(id, name, categoryEnum, aviability, price, colour, 1);
        this.colour = colour;
    }

    public String getColour() {
        return colour;
    }

    public void setColour(String colour) {
        this.colour = colour;
    }

    @Override
    public String getHeaders() {
        return super.getHeaders() + "colour";
    }

    @Override
    public String toCSV() {
        return super.toCSV() + colour;
    }

}

    

File: Python/lab14/goods/Sketchbook.py
from goods.Good import Good


class Sketchbook(Good):
    def __init__(self, name, category, price, colour, pages):
        super().__init__(name, category, price, colour, pages)
        self.colour = colour

    def __str__(self):
        return ", Name:" + str(self.name) + " Category: " + str(self.category) + ", Price:" + str(
            self.price) + ", Colour:" + str(self.colour) + ", Pages:" + str(self.pages)

